### PR TITLE
Update Channel Terminated Message cross chain command

### LINKED ISSUE #7656: Update Channel Terminated Message cross chain command
### Description

- Update command schema
```js
channelTerminatedCCMParamsSchema = {
    "type": "object",
    "required": ["stateRoot", "inboxSize"],
    "properties": {
        "stateRoot": {
            "dataType": "bytes",
            "length": HASH_LENGTH,
            "fieldNumber": 1
        },
        "inboxSize": {
            "dataType": "uint32",
            "fieldNumber": 2
        }
    }
}
```
- Update execute method
```python
def execute(ccu: Transaction, ccm: CCM) -> None:
    if isLive(ccm.sendingChainID):
        createTerminatedStateAccount(ccm.sendingChainID)
```

### Acceptance Criteria

- Should have all the unit tests

### Additional Information

- Related PR https://github.com/LiskHQ/lips/pull/148
- Related LIP https://github.com/LiskHQ/lips/blob/main/proposals/lip-0049.md#channel-terminated-message-1

### PR DESCRIPTION
**Pull request by Phanco**
### What was the problem?

This PR resolves #7656

### How was it solved?

Created a new `BaseCCChannelTerminatedCommand` inside `base/cc_commands`

### How was it tested?

Unit Test Updated accordingly

### CODE REVIEW

File: framework/src/modules/interoperability/base/cc_commands/channel_terminated.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+				context,
+				context.ccm.sendingChainID,
+				channelTerminatedCCMParams.stateRoot,
+			);
+			await interoperabilityStore.createTerminatedOutboxAccount(
+				context.ccm.sendingChainID,
+				ccmParamsChannel.outbox.root,
+				ccmParamsChannel.outbox.size,
+				channelTerminatedCCMParams.inboxSize,
+			);
```
Comment (ishantiw) [inline framework/src/modules/interoperability/base/cc_commands/channel_terminated.ts:49]: This logic is removed from the recent update, please refer to https://github.com/LiskHQ/lips/blob/main/proposals/lip-0049.md#channel-terminated-message-1, I have also updated the issue

File: framework/src/modules/interoperability/schemas.ts [hunk 2]
```diff
@@ -345,7 +346,19 @@ export const registrationCCMParamsSchema = {
...
+			minLength: HASH_LENGTH,
+			maxLength: HASH_LENGTH,
+			fieldNumber: 1,
+		},
+		inboxSize: {
+			dataType: 'uint32',
+			fieldNumber: 2,
+		},
+	},
 };
```
Comment (ishantiw) [inline framework/src/modules/interoperability/schemas.ts:362]: now we have an empty schema

**Review by ishantiw — CHANGES_REQUESTED** [review]
maybe its better to rename, `base_classes` or `common` to be more precise and also move other base classes within it that are present within `interoperability` folder?

**Comment by Phanco** [comment]
> maybe its better to rename, `base_classes` or `common` to be more precise and also move other base classes within it that are present within `interoperability` folder?

Since those classes starts with `Base` so I would go probably for `base_classes`

**Comment by Phanco** [comment]
@ishantiw that involves quite a number of file changes if I move `base_*` files to `base_classes` folder. Is it okay if I do it in another PR?
image

**Comment by ishantiw** [comment]
lets refactor this later in another issue,
https://github.com/LiskHQ/lisk-sdk/pull/7696#issuecomment-1294882751

Meanwhile use `base_cc_commands` folder instead of just `base` and later we can rename it to `base_classes` and move everything else.

**Review by ishantiw — COMMENTED** [review]
@Phanco lgtm, maybe we can move test to one common file as @sitetester suggests. You can follow similar approach as we have in https://github.com/LiskHQ/lisk-sdk/pull/7700

**Comment by Phanco** [comment]
> @Phanco lgtm, maybe we can move test to one common file as @sitetester suggests. You can follow similar approach as we have in #7700

Took reference on #7700 and updated test case, thanks :D

File: framework/src/modules/interoperability/sidechain/cc_commands/channel_terminated.ts [hunk 3]
```diff
@@ -13,30 +13,11 @@
...
-			throw new Error('CCM to execute channel terminated cross chain command is missing.');
-		}
-		await interoperabilityInternalMethod.createTerminatedStateAccount(
-			context,
-			context.ccm.sendingChainID,
-		);
-	}
-
+// https://github.com/LiskHQ/lips/blob/main/proposals/lip-0049.md#channel-terminated-message-1
+export class SidechainCCChannelTerminatedCommand extends BaseCCChannelTerminatedCommand {
```
[the remark this answers is not in the document]
Comment (Phanco) [reply framework/src/modules/interoperability/sidechain/cc_commands/channel_terminated.ts:20]: I saw quite a few `getInteroperabilityInternalMethod` are using `context: StoreGetter | ImmutableStoreGetter`. 

Yeah I think need some inspiration from @ishantiw

File: framework/test/unit/modules/interoperability/base_cc_commands/channel_terminated.spec.ts [hunk 4]
```diff
@@ -66,14 +68,21 @@ describe('MainchainCCChannelTerminatedCommand', () => {
 		ccMethodsMap,
 	);
 	mainchainInteroperabilityInternalMethod.createTerminatedStateAccount = createTerminatedStateAccountMock;
+	mainchainInteroperabilityInternalMethod.isLive = jest.fn().mockResolvedValue(false);
```
[the remark this answers is not in the document]
Comment (Phanco) [reply framework/test/unit/modules/interoperability/base_cc_commands/channel_terminated.spec.ts]: Done :D

File: framework/src/modules/interoperability/sidechain/cc_commands/channel_terminated.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ishantiw) [reply framework/src/modules/interoperability/sidechain/cc_commands/channel_terminated.ts:20]: basically we want to remove `ImmutableStoreGetter` in future and use `context: StoreGetter | ImmutableStoreGetter` but lets tackle it later when we have more clear picture. I think we should think about i and do it after https://github.com/LiskHQ/lisk-sdk/issues/7719

**Review by ishantiw — APPROVED** [review]

**Pull request merged**